src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -230,5 +230,35 @@ private enum WM : uint
             CLOSE = 0x0010
         }
 #endif
+
+        /// <summary>Gets current principal of a thread</summary>
+        /// <returns><see cref="Thread.CurrentPrincipal"/>; null if <see cref="Thread.CurrentPrincipal"/> getter throws <see cref="PlatformNotSupportedException"/>.</returns>
+        public static System.Security.Principal.IPrincipal GetCurrentThreadPrincipal()
+        {
+            try
+            {
+                return Thread.CurrentPrincipal;
+            }
+            catch (PlatformNotSupportedException) //E.g. Mono.WASM
+            {
+                return null;
+            }
+        }
+
+        /// <summary>Sets thread current principal</summary>
+        /// <param name="principal">Value to set</param>
+        /// <remarks>
+        /// This method sets <see cref="Thread.CurrentPrincipal"/>.
+        /// In case <see cref="Thread.CurrentPrincipal"/> setter throws <see cref="PlatformNotSupportedException"/> and <paramref name="principal"/> was null, this method silently does nothing.
+        /// </remarks>
```

Comment:
Are remarks visible in intellisense? If not, pretty much the same here as the other comments.